### PR TITLE
Valid_move? for the Knight

### PR DESCRIPTION
**Pull request by hugolmendez**
UPDATE: Refactored code.  Eliminated the private section.  Initially the tests were not passing, but the additional method was not needed.  Also added tests for invalid moves.  Thanks, @CodeFlowed :-)
UPDATED: Corrected all my tests. Thanks to @leamerluck I found out that the logic I was using for my coordinates was wrong.  I could have won every game if I had it my way.   :)     Still, I hope it can be reviewed and make sure it's good.  

Added the valid_move? logic method for the Knight piece. 
All tests passed, so I think is ready.   This didn't require the is_obstructed? method. 
Please review and let me know if something needs to be changed or if you give it a :+1:  .

### CODE REVIEW

File: test/models/knight_test.rb [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+        assert_equal expected, actual
+
+        #tests white moves from starting postion
+        actual = knight.valid_move?(1, 6)
+        assert_equal expected, actual
+
+        #tests white moves from starting postion
+        actual = knight.valid_move?(4, 7)
+        assert_equal expected, actual
+    end
```
Comment (CodeFlowed) [inline test/models/knight_test.rb]: @hugolmendez Nice work on the tests here. Sorry for being a little nitpicky, but can you also include a test for an invalid knight move?

File: app/models/knight.rb [hunk 2]
```diff
@@ -1,3 +1,15 @@
...
+  def valid_move?(new_x, new_y)
+    false if x_position == new_x
+    false if y_position == new_y
+    right_move?(new_x, new_y)
+  end
+
+  private
+
+  def right_move?(new_x, new_y)
+    (x_position - new_x).abs + (y_position - new_y).abs == 3
```
Comment (CodeFlowed) [inline app/models/knight.rb]: Liking this! :smile:

**Comment by leamerluck** [comment]
Other than my comment/question above. This looks good to me :+1:

**Comment by CodeFlowed** [comment]
@leamerluck Yeah, I was wondering, too, about the extra checks here and that are also in my code for the queen and bishop valid moves. Do we already have a way to validate or sanitize the parameters that are sent to the methods?

That said, we can refactor the code and remove unnecessary tests later, IMO. I'll give it a :+1: for merging.

**Pull request merged**